noir-r1cs/benches/bench.rs

```diff
@@ -22,29 +22,15 @@ fn prove_poseidon_1000(bencher: Bencher) {
         .join("poseidon-1000.nps");
     let scheme: NoirProofScheme = read(&path).unwrap();
 
-    // Run nargo compile
-    let status = std::process::Command::new("nargo")
-        .arg("compile")
-        .current_dir("../noir-examples/poseidon-rounds")
-        .status()
-        .expect("Running nargo compile");
-    if !status.success() {
-        panic!("Failed to run nargo compile");
-    }
+    let crate_dir: &Path = "../noir-examples/poseidon-rounds".as_ref();
 
-    // Run nargo execute
-    let status = std::process::Command::new("nargo")
-        .arg("execute")
-        .current_dir("../noir-examples/poseidon-rounds")
-        .status()
-        .expect("Running nargo execute");
-    if !status.success() {
-        panic!("Failed to run nargo execute");
-    }
+    compile_workspace(crate_dir).expect("Compiling workspace");
+
+    let program_path = crate_dir.join("target/basic.json");
+    let witness_path = crate_dir.join("Prover.toml");
+
+    let witness_map = execute_program_witness(program_path, witness_path).unwrap();
 
-    let witness_file_path = &PathBuf::from("../noir-examples/poseidon-rounds/target/basic.gz");
-    let mut witness_stack = deserialize_witness_stack(witness_file_path).unwrap();
-    let witness_map: WitnessMap<NoirFieldElement> = witness_stack.pop().unwrap().witness;
     bencher.bench(|| black_box(&scheme).prove(black_box(&witness_map)));
```

Comment:
A prove should start with a `.nps` and `.toml` file and end with a `.np` file. Nothing more nothing less. Here file reading and witness computation are excluded. It's fine to have such a benchmark (doing only partial proof) but we also need a benchmark that does an end-to-end proof.